scripts/dwi_estimation_error_analysis.py

```diff
@@ -0,0 +1,207 @@
+    gpm = EddyMotionGPR(
+        kernel=SphericalKriging(a=2.15, lambda_s=120),
+        alpha=50,
+        optimizer=None,
+    )
+
+    X = gtab[~gtab.b0s_mask].bvecs
+    y = signal[~gtab.b0s_mask]
+
+    rkf = RepeatedKFold(n_splits=cv, n_repeats=120 // cv)
+    scores = cross_val_score(gpm, X, y, scoring="neg_root_mean_squared_error", cv=rkf)
+    return scores
+
+
+def _build_arg_parser() -> argparse.ArgumentParser:
+    """
+    Build argument parser for command-line interface.
+
+    Returns
+    -------
+    :obj:`~argparse.ArgumentParser`
+        Argument parser for the script.
+
+    """
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument(
+        "hsph_dirs",
+        help="Number of diffusion gradient-encoding directions in the half sphere",
+        type=int,
+    )
+    parser.add_argument("bval_shell", help="Shell b-value", type=float)
+    parser.add_argument("S0", help="S0 value", type=float)
+    parser.add_argument("--evals1", help="Eigenvalues of the tensor", nargs="+", type=float)
+    parser.add_argument("--snr", help="Signal to noise ratio", type=float)
+    parser.add_argument("--repeats", help="Number of repeats", type=int, default=5)
+    parser.add_argument(
+        "--kfold", help="Number of directions to leave out/predict", nargs="+", type=int
+    )
+    return parser
+
+
+def _parse_args(parser: argparse.ArgumentParser) -> argparse.Namespace:
+    """
+    Parse command-line arguments.
+
+    Parameters
+    ----------
+    parser : :obj:`~argparse.ArgumentParser`
+        Argument parser for the script.
+
+    Returns
+    -------
+    :obj:`~argparse.Namespace`
+        Parsed arguments.
+    """
+    return parser.parse_args()
+
+
+def main() -> None:
+    """Main function for running the experiment and plotting the results."""
+    parser = _build_arg_parser()
+    args = _parse_args(parser)
+
+    # create eigenvectors for a single fiber
+    evecs = testsims.create_single_fiber_evecs()
+
+    # Create a gradient table for a single-shell
+    gtab = testsims.create_single_shell_gradient_table(args.hsph_dirs, args.bval_shell)
+
+    # Use Scikit-learn cross validation
+    scores = defaultdict(list, {})
+
+    for n in args.kfold:
+        for i in range(args.repeats):
+            cv_scores = -1.0 * cross_validate(gtab, args.S0, args.evals1, evecs, args.snr, n)
+            scores["rmse"] += cv_scores.tolist()
+            scores["repeat"] += [i] * len(cv_scores)
+            scores["n_folds"] += [n] * len(cv_scores)
+
+        print(f"Finished {n}-fold cross-validation")
+    
+    scores_df = pd.DataFrame(scores)
+    scores_df.to_csv("cv_scores.tsv", sep="\t", index=None, na_rep="n/a")
+
+    grouped = scores_df.groupby(["n_folds"])
+    print(grouped[["rmse"]].mean())
+    print(grouped[["rmse"]].std())
+
+    # Plot - import plot_* from eddymotion.viz.signals
```

Comment:
Keeping the code here for future. Happy to wait for these plots to happen, but also ready to merge and work the visualizations elsewhere.